src/routes.py

```diff
@@ -130,18 +116,12 @@ def topic_page(topic_id: int) -> str | Response:
 @bp.route("/topics/<int:topic_id>/posts/create", methods=["POST", "GET"])
 def create_post(topic_id: int) -> str | Response:  # noqa: CFQ004
     """Handle post creation page."""
-    session_id = request.cookies.get("session_id")
-    user_session = None
-    if session_id is not None:
-        user_session = UserSession.get_user_session_by_session_id(session_id)
-
-    if user_session is None:
+    if (current_user := get_current_user()) is None:
         return redirect(url_for("routes.login"))
-    current_user = user_session.user
     form = PostForm()
     if not (topic := Topic.get(topic_id)):
         return render_template("404.html", current_user=current_user)
-    if topic and form.validate_on_submit():
+    if form.validate_on_submit():
```

Comment:
Seems like this change is not related to the scope of the ticket. I'll revert it before merge, and let's make this change in the another [ticket](https://github.com/fenya123/forum123/issues/76) I've created for this.